src/spawn_players.c

```diff
@@ -531,100 +531,75 @@ void spawn_players_gp_one_player(f32* arg0, f32* arg1, f32 arg2) {
-            spawn_player(gPlayerTwo, 1, arg0[D_80165270[1]], arg1[D_80165270[1]] + 250.0f, arg2, 32768.0f,
-                         chooseCPUPlayers[0], PLAYER_EXISTS | PLAYER_STAGING | PLAYER_START_SEQUENCE | PLAYER_CPU);
-            spawn_player(gPlayerThree, 2, arg0[D_80165270[3]], arg1[D_80165270[2]] + 250.0f, arg2, 32768.0f,
-                         chooseCPUPlayers[1], PLAYER_EXISTS | PLAYER_STAGING | PLAYER_START_SEQUENCE | PLAYER_CPU);
-            spawn_player(gPlayerFour, 3, arg0[D_80165270[2]], arg1[D_80165270[3]] + 250.0f, arg2, 32768.0f,
-                         chooseCPUPlayers[2], PLAYER_EXISTS | PLAYER_STAGING | PLAYER_START_SEQUENCE | PLAYER_CPU);
-            spawn_player(gPlayerFive, 4, arg0[D_80165270[5]], arg1[D_80165270[4]] + 250.0f, arg2, 32768.0f,
-                         chooseCPUPlayers[3], PLAYER_EXISTS | PLAYER_STAGING | PLAYER_START_SEQUENCE | PLAYER_CPU);
-            spawn_player(gPlayerSix, 5, arg0[D_80165270[4]], arg1[D_80165270[5]] + 250.0f, arg2, 32768.0f,
-                         chooseCPUPlayers[4], PLAYER_EXISTS | PLAYER_STAGING | PLAYER_START_SEQUENCE | PLAYER_CPU);
-            spawn_player(gPlayerSeven, 6, arg0[D_80165270[7]], arg1[D_80165270[6]] + 250.0f, arg2, 32768.0f,
-                         chooseCPUPlayers[5], PLAYER_EXISTS | PLAYER_STAGING | PLAYER_START_SEQUENCE | PLAYER_CPU);
-            spawn_player(gPlayerEight, 7, arg0[D_80165270[6]], arg1[D_80165270[7]] + 250.0f, arg2, 32768.0f,
-                         chooseCPUPlayers[6], PLAYER_EXISTS | PLAYER_STAGING | PLAYER_START_SEQUENCE | PLAYER_CPU);
+            for (int i = 1; i < NUM_PLAYERS; i++) {
+                spawn_player(&gPlayers[i], i, arg0[D_80165270[i]], arg1[D_80165270[i]] + 250.0f, arg2, 32768.0f,
+                            chooseCPUPlayers[i - 1], PLAYER_EXISTS | PLAYER_STAGING | PLAYER_START_SEQUENCE | PLAYER_CPU);
+            }
         }
         D_80164A28 = 1;
     }
     func_80039AE4();
 }
 
 void spawn_players_versus_one_player(f32* arg0, f32* arg1, f32 arg2) {
-    spawn_player(gPlayerFour, 3, arg0[2], arg1[2], arg2, 32768.0f, gCharacterSelections[0],
-                 PLAYER_START_SEQUENCE | PLAYER_CPU);
-    spawn_player(gPlayerFive, 4, arg0[3], arg1[3], arg2, 32768.0f, gCharacterSelections[0],
-                 PLAYER_START_SEQUENCE | PLAYER_CPU);
-    spawn_player(gPlayerSix, 5, arg0[4], arg1[4], arg2, 32768.0f, gCharacterSelections[0],
-                 PLAYER_START_SEQUENCE | PLAYER_CPU);
-    spawn_player(gPlayerSeven, 6, arg0[5], arg1[5], arg2, 32768.0f, gCharacterSelections[0],
-                 PLAYER_START_SEQUENCE | PLAYER_CPU);
-    spawn_player(gPlayerEight, 7, arg0[6], arg1[6], arg2, 32768.0f, gCharacterSelections[0],
-                 PLAYER_START_SEQUENCE | PLAYER_CPU);
+    for (int i = 3; i < NUM_PLAYERS; i++) {
+        spawn_player(&gPlayers[i], i, arg0[i], arg1[i], arg2, 32768.0f, gCharacterSelections[i],
```

Comment:
arg0[i] is supposed to equal 2 here. But i starts at 3. I don't think this is func equiv